### PR TITLE
Add 'Running Asynchronous Tasks' guide closes #320

### LINKED ISSUE #320: Add Task.Supervisor guides
// @jeregrine

### PR DESCRIPTION
**Pull request by jsncmgs1**

### CODE REVIEW

File: bonus_guides/X_task_supervisor.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+our async task, it doesn't crash the calling process. To do this, we need a
+separate supervisor that can supervise our async process. We can use the
+`HelloPhoenix.TaskSupervisor` that we created earlier for this, as we will see
+below.
+
+Let's prove that we can send a task to our supervisor that is completely
+independent of the calling process, allowing the caller to continue running if
+the task blows up.
+
+In the new processes, we will crash it with `1/0`, to make sure our request still
```
Comment (Gazler) [inline bonus_guides/X_task_supervisor.md]: Perhaps we could add

 > (which will raise `(ArithmeticError) bad argument in arithmetic expression`)

File: bonus_guides/X_task_supervisor.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+`async_nolink/2` accepts the name of a supervisor as the first argument, we
+passed in the name of the supervisor we specified in `lib/hello_phoenix.ex`.
+The next argument is an anonymous function that will become a task supervised
+by the passed in supervisor.  As the name suggested, this task will not be
+linked to the calling process, allowing our request to finish when our task
+fails.
+
+When we spin up our server and visit our `"/"` route at `localhost:4000`, if
```
Comment (Gazler) [inline bonus_guides/X_task_supervisor.md]: Can we change:

 > if you watch the server you'll see the request finish as normal

to:

> the request finishes and responds as normal - followed...

File: bonus_guides/X_task_supervisor.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+you watch the server you'll see the request finish as normal - followed by an
+arithmetic error after a few seconds. Since the tasks aren't linked, we can
+"fire and forget" any task using the other supervisor, and our controller is
```
Comment (Gazler) [inline bonus_guides/X_task_supervisor.md]: "fire and forget" has been mentioned several times above. Perhaps we could say "execute any task"?

File: bonus_guides/X_task_supervisor.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+able to send a response regardless of the error.
+
+It's worth noting that if these tasks fail, they won't be retried.  For systems
+that require persistent tasks, consider using
+[GenStage](https://hexdocs.pm/gen_stage/Experimental.GenStage.html)
```
Comment (Gazler) [inline bonus_guides/X_task_supervisor.md]: People may get the idea that GenStage is a worker library with retries.

Perhaps we could say:

 > consider building a solution using GenStage

instead?

File: bonus_guides/X_task_supervisor.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
+# Running Asynchronous Tasks
+
+Elixir makes concurrency dead simple right out of the box, allowing us to
```
Comment (Gazler) [inline bonus_guides/X_task_supervisor.md]: I'd remove this first sentence and just start with "There are many instances..."

File: bonus_guides/X_task_supervisor.md (hunk 4 above)
Comment (jsncmgs1) [inline bonus_guides/X_task_supervisor.md]: Sounds good, I'll get these updated later tonight.

**Comment by jsncmgs1** [comment]
@Gazler I made those changes. Thanks for the review!

**Comment by Gazler** [comment]
Thanks <3

**Pull request merged**